src/java_bytecode/java_types.h

```diff
@@ -313,6 +320,67 @@ inline const typet &java_generic_class_type_bound(size_t index, const typet &t)
   return gen_type.subtype();
 }
 
+/// Class to hold a class that is implicitly generic, e.g., an inner
+/// class of a generic outer class or a derived class of a generic base
+/// class. Extends the java class type.
+class java_implicitly_generic_class_typet:public java_class_typet
+{
+public:
+  typedef std::vector<java_generic_parametert> implicit_generic_typest;
+
+  static const java_implicitly_generic_class_typet &from_java_class_type(
+    java_class_typet &class_type,
+    implicit_generic_typest implicit_generic_types)
+  {
+    class_type.set(ID_C_java_implicitly_generic_class_type, true);
+    java_implicitly_generic_class_typet &implicitly_generic_class_type =
+      static_cast<java_implicitly_generic_class_typet &>(class_type);
+    for(const java_generic_parametert &param : implicit_generic_types)
+    {
+      implicitly_generic_class_type.implicit_generic_types().push_back(param);
+    }
+    return implicitly_generic_class_type;
+  }
+
+  const implicit_generic_typest &implicit_generic_types() const
+  {
+    return (
```

Comment:
in general no "(" ")" around return value

Reply:
The parentheses are for casting, I changed the formatting.